app/src/main/kotlin/com/simplemobiletools/clock/extensions/Context.kt

```diff
@@ -266,20 +272,53 @@ fun Context.formatTo12HourFormat(showSeconds: Boolean, hours: Int, minutes: Int,
     return "${formatTime(showSeconds, false, newHours, minutes, seconds)} $appendable"
 }
 
-fun Context.getNextAlarm(): String {
-    val milliseconds = (getSystemService(Context.ALARM_SERVICE) as AlarmManager).nextAlarmClock?.triggerTime ?: return ""
-    val calendar = Calendar.getInstance()
-    val isDaylightSavingActive = TimeZone.getDefault().inDaylightTime(Date())
-    var offset = calendar.timeZone.rawOffset
-    if (isDaylightSavingActive) {
-        offset += TimeZone.getDefault().dstSavings
+fun Context.getClosestEnabledAlarmString(result: (String) -> Unit) {
+    getEnabledAlarms { enabledAlarms ->
+        if (enabledAlarms == null) {
+            result.invoke("")
```

Comment:
you can just use result(""), if it isnt nullable

Reply:
done